### PR TITLE
AMLS-3873 | Prevent changing previous notifications when comms change

### PR DESCRIPTION
**Pull request by drewmiley**
Creating this PR to review work in progress

## Related / Dependant PRs?

https://github.com/hmrc/amls-notification/pull/41

## How Has This Been Tested?

Locally manually.

## Checklist

- [ ] Breaking change (fix or feature that would cause existing functionality to change).
- [x] Build passing.
- [ ] Unit tests have full coverage.
- [ ] Unit test approach and implementation has been reviewed with QA (testers).
- [x] Requires acceptance test run.
- [x] Appropriate labels added.

### CODE REVIEW

File: test/views/notifications/v1/minded_to_revoke_flattenSpec.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/minded_to_revoke_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: test/views/notifications/v1/no_longer_minded_to_reject_flattenSpec.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/no_longer_minded_to_reject_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: test/views/notifications/v1/no_longer_minded_to_revoke_flattenSpec.scala [hunk 3]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/no_longer_minded_to_revoke_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: test/views/notifications/v1/rejection_reasons_flattenSpec.scala [hunk 4]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/rejection_reasons_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: test/views/notifications/v1/revocation_reasons_flattenSpec.scala [hunk 5]
```diff
@@ -0,0 +1,44 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/revocation_reasons_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: test/views/notifications/v1/your_messages_flattenSpec.scala [hunk 6]
```diff
@@ -0,0 +1,104 @@
+/*
```
Comment (drewmiley) [inline test/views/notifications/v1/your_messages_flattenSpec.scala]: Remove when un-versioned templates are deleted

File: app/controllers/NotificationController.scala [hunk 7]
```diff
@@ -76,11 +78,11 @@ class NotificationController @Inject()(
       val previousRecords: Seq[NotificationRow] = (for {
         amls <- refNumber
       } yield records.filter(_.amlsRegistrationNumber != amls)) getOrElse Seq()
-      Ok(views.html.notifications.your_messages(businessName, currentRecords, previousRecords))
+      Ok(views.html.notifications.v1.your_messages(NotificationListParams(businessName, currentRecords, previousRecords)))
```
[the remark this answers is not in the document]
Comment (JGRobinson) [reply app/controllers/NotificationController.scala]: Yep - Your_messages shouldn't be versioned, its not a template

File: upversion-templates.sh [hunk 8]
```diff
@@ -0,0 +1,72 @@
+minorflag=false
+if [ "$1" = "minor" ]; then
+    minorflag=true
+fi
+
+packages=(./app/views/notifications/*/)
+
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply upversion-templates.sh]: Added

File: upversion-templates.sh [hunk 9]
```diff
@@ -0,0 +1,72 @@
+minorflag=false
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply upversion-templates.sh]: Added

File: app/models/notifications/NotificationRow.scala [hunk 10]
```diff
@@ -31,6 +31,7 @@ case class NotificationRow(
                             receivedAt: DateTime,
                             isRead: Boolean,
                             amlsRegistrationNumber: String,
+                            templatePackageVersion: String = "v1m0",
```
[the remark this answers is not in the document]
Comment (slavvco) [reply app/models/notifications/NotificationRow.scala]: Done

File: upversion-templates.sh [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+if [ "$1" = "minor" ]; then
+    minorflag=true
+fi
+
+packages=(./app/views/notifications/*/)
```
[the remark this answers is not in the document]
Comment (drewmiley) [reply upversion-templates.sh]: Added

**Pull request merged**